src/main/java/uk/gov/companieshouse/web/accounts/controller/smallfull/ApprovalController.java

```diff
@@ -119,6 +129,15 @@ public String postApproval(@PathVariable String companyNumber,
                 return getTemplateName();
             }
 
+            String uri = TRANSACTIONS_URI.expand(transactionId).toString();
+            Transaction transaction = apiClientService.getApiClient().transactions().get(uri).execute().getData();
```

Comment:
Maybe it is worth obscuring this logic behind `TransactionService` - it feels a little odd for this controller to be aware of `ApiErrorResponseException`'s

Reply:
@BSaunders95 By 'obscuring this logic behind TransactionService', do you mean move the logic down into TransactionService?

Reply:
Moved the logic to transaction service impl class